src/pemfile.rs

```diff
@@ -37,6 +43,56 @@ pub enum Item {
     Crl(CertificateRevocationListDer<'static>),
 }
 
+/// Errors that may arise when parsing the contents of a PEM file
+pub enum Error {
```

Comment:
Could this change replacing the `io::Error` usage with a dedicated enum be separated out from this commit? It doesn't seem related to the `read_one_from_slice` change and isn't mentioned in the commit message.

Reply:
this new `Error` type is only used in the new `read_one_from_slice` API. I didn't change the signature of the existing public API as that would be a breaking change.

Reply:
The `main` branch already has breaking changes relatively to released versions, so it's probably fine to break it more if we feel that provides the best API.